TfsMigrate.Processors.WorkItemTracking/WorkItemTrackingProcessor.cs

```diff
@@ -440,69 +449,107 @@ private Task LoadUsersAsync ( MigrationContext context, IEnumerable<UserSettings
                 var isChild = relation.IsChild();
                 var isParent = !isChild && relation.IsParent();
                 var isRelated = !isChild && !isParent && relation.IsRelated();
+                var isAttachment = relation.IsAttachment();
 
                 //Not a relation we care about                
-                if (!isChild && !isParent && !isRelated)
+                if (!isChild && !isParent && !isRelated && !isAttachment)
                     continue;
 
-                //Get the related item information
-                var relatedId = relation.GetRelatedId();
-                if (relatedId <= 0)
-                    continue;
-
-                var related = context.GetMigratedWorkItem(relatedId);
-                
-                if (isChild)
+                if (isAttachment)

```

Comment:
Other than enumerating the relations this isn't really associated with the relation processing that is being done here.
Move the attachment logic into its own method and then update the calling code to call it after MigrateWorkItemRelationsAsync is called. This would reduce the amount of conditional logic being used here.